vericoding/processing/file_processor.py

```diff
@@ -0,0 +1,277 @@
+                fix_prompt = prompt_loader.format_prompt(
+                    "fix_verification",
+                    code=current_code,
+                    errorDetails=error_details,
+                    iteration=iteration,
+                )
+
+                try:
+                    fix_response = call_llm_api(config, fix_prompt)
+                    fixed_code = extract_code(config, fix_response)
+
+                    # Verify that all specifications are still preserved
+                    if config.strict_spec_verification and not verify_spec_preservation(
+                        config, original_code, fixed_code
+                    ):
+                        thread_safe_print(
+                            "    ⚠️  Warning: Specifications were modified during fix. Restoring original specifications..."
+                        )
+                        fixed_code = restore_specs(config, original_code, fixed_code)
+
+                    current_code = fixed_code
+                    thread_safe_print(f"    Generated fix for iteration {iteration}")
+                except Exception as e:
+                    thread_safe_print(f"    ✗ Failed to generate fix: {str(e)}")
+                    break
+
+        if success:
+            thread_safe_print(
+                f"  ✓ Successfully generated and verified: {output_path.name}"
+            )
+            return ProcessingResult(
+                success=True,
+                file=str(relative_path),
+                output=str(output_path),
+                error=None,
+                has_bypass=False,
+            )
+        else:
+            error_msg = (
+                last_verification.error
+                if last_verification
+                else "Unknown verification error"
+            )
+            thread_safe_print(
+                f"  ✗ Failed to verify after {config.max_iterations} iterations: {error_msg[:200] if error_msg else 'Unknown error'}..."
+            )
+            return ProcessingResult(
+                success=False,
+                file=str(relative_path),
+                output=str(output_path),
+                error=error_msg,
+                has_bypass=False,
+            )
+
+    except Exception as e:
+        thread_safe_print(f"✗ Failed: {Path(file_path).name} - {str(e)}")
+        return ProcessingResult(
+            success=False,
+            file=str(relative_path)
+            if "relative_path" in locals()
+            else Path(file_path).name,
+            error=str(e),
+            output=None,
+            has_bypass=False,
+        )
+
+
+def process_files_parallel(
+    config: ProcessingConfig, prompt_loader: PromptLoader, spec_files: list[str]
+) -> list[ProcessingResult]:
+    """Process files in parallel using ThreadPoolExecutor."""
+    results = []
+    completed_count = 0
+    total_files = len(spec_files)
+
+    print(
+        f"Processing {total_files} files using {config.max_workers} parallel workers..."
```

Comment:
later we should upgrade this too, but not for now. maybe morph is well suited to this part